render/reply/formatter_test.go

```diff
@@ -0,0 +1,240 @@
+package reply
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"net/http/httptest"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/lomik/graphite-clickhouse/finder"
+	"github.com/lomik/graphite-clickhouse/helper/client"
+	"github.com/lomik/graphite-clickhouse/helper/point"
+	"github.com/lomik/graphite-clickhouse/pkg/alias"
+	"github.com/lomik/graphite-clickhouse/render/data"
+)
+
+var results = []client.Metric{
+	{
+		Name:           "test.metric1",
+		PathExpression: "test.*",
+		StartTime:      1688990040,
+		StepTime:       60,
+		StopTime:       1688990520,
+		Values: func() []float64 {
+			temp := emptyValues(8)
+			temp[2] = 3
+			return temp
+		}(),
+	},
+	{
+		Name:           "test.metric2",
+		PathExpression: "test.*",
+		StartTime:      1688990040,
+		StepTime:       60,
+		StopTime:       1688990520,
+		Values:         emptyValues(8),
+	},
+	{
+		Name:           "test.metric3",
+		PathExpression: "test.*",
+		StartTime:      1688990040,
+		StepTime:       60,
+		StopTime:       1688990520,
+		Values:         emptyValues(8),
+	},
+}
+
+func TestFormatterReply(t *testing.T) {
+	formatters := []struct {
+		impl   Formatter
+		name   string
+		format client.FormatType
+	}{
+		{&V3PB{}, "v3pb", client.FormatPb_v3},
+		{&V2PB{}, "v2pb", client.FormatPb_v2},
+		{&JSON{}, "json", client.FormatJSON},
+		{&Pickle{}, "pickle", client.FormatPickle},
+	}
+	tests := []struct {
+		name  string
+		input data.CHResponses
+		// result when CHResponse.AppendOutEmptySeries is false
+		expected0 []client.Metric
+		// result when CHResponse.AppendOutEmptySeries is true
+		expected1 []client.Metric
+	}{
+		{
+			name:      "no index found",
+			input:     data.EmptyResponse(),
+			expected0: []client.Metric{},
+			expected1: []client.Metric{},
+		},
+		{
+			name: "three indexed items found, but only one has points",
+			input: prepareCHResponses(1688990000, 1688990460,
+				[][]byte{[]byte("test.metric1"), []byte("test.metric2"), []byte("test.metric3")},
+				map[string][]point.Point{
+					"test.metric1": {{Value: 3, Time: 1688990160, Timestamp: 1688990204}},
+				},
+			),
+			expected0: results[:1],
+			expected1: results,
+		},
+		{
+			name: "three indexed items found, no points in all",
```

Comment:
Mya be tests for all 3 metrics (`test.metric1` with points and other with NaN) ?